ion/textwriter.go

```diff
@@ -411,21 +414,12 @@ func (w *textWriter) writeFieldName(api string) error {
 }
 
 // writeAnnotations writes out the annotations for a value.
-func (w *textWriter) writeAnnotations(api string) error {
+func (w *textWriter) writeAnnotations() error {
 	as := w.annotations
 	w.annotations = nil
 
-	var text string
 	for _, a := range as {
-		if a.Text != nil {
-			text = *a.Text
-		} else if a.LocalSID != SymbolIDUnknown {
-			text = fmt.Sprintf("$%v", a.LocalSID)
-		} else {
-			return &UsageError{api, "invalid annotation symbol token"}
-		}
-
-		if err := writeSymbol(text, w.out); err != nil {
+		if err := writeSymbol(a.toText(), w.out); err != nil {
```

Comment:
If I read the code correctly, when we have ```a.Text``` is nil and ```a.LocalSID``` is Unknown, then ```toText()``` will return an empty string. And this empty text is sent to ```writeSymbol```.
I think if ```a.Text``` is nil and ```a.LocalSID``` is Unknown, error should be returned.

Reply:
Addressed.